crates/ruff_python_formatter/src/expression/string/docstring.rs

```diff
@@ -273,34 +292,39 @@ impl<'ast, 'buf, 'fmt, 'src> DocstringLinePrinter<'ast, 'buf, 'fmt, 'src> {
         {
             return self.print_one(&line.as_output());
         }
-        match self.code_example.add(line) {
-            CodeExampleAddAction::Print { original } => self.print_one(&original.as_output())?,
-            CodeExampleAddAction::Kept => {}
-            CodeExampleAddAction::Reset { code, original } => {
-                for codeline in code {
-                    self.print_one(&codeline.original.as_output())?;
+        self.code_example.add(line, &mut self.action_queue);
+        self.run_action_queue()
+    }
+
+    /// Process any actions in this printer's queue until the queue is empty.
+    fn run_action_queue(&mut self) -> FormatResult<()> {
+        while let Some(action) = self.action_queue.pop_front() {
+            match action {
+                CodeExampleAddAction::Print { original } => {
+                    self.print_one(&original.as_output())?;
                 }
-                self.print_one(&original.as_output())?;
-            }
-            CodeExampleAddAction::Format { mut kind, original } => {
-                let Some(formatted_lines) = self.format(kind.code())? else {
-                    // If formatting failed in a way that should not be
-                    // allowed, we back out what we're doing and print the
-                    // original lines we found as-is as if we did nothing.
-                    for codeline in kind.code() {
+                CodeExampleAddAction::Kept => {}
+                CodeExampleAddAction::Reset { code } => {
+                    for codeline in code {
                         self.print_one(&codeline.original.as_output())?;
                     }
-                    if let Some(original) = original {
-                        self.print_one(&original.as_output())?;
-                    }
-                    return Ok(());
-                };
+                }
+                CodeExampleAddAction::Format { mut kind } => {
+                    let Some(formatted_lines) = self.format(kind.code())? else {
+                        // Since we've failed to emit these lines, we need to put
+                        // them back in the queue but have them jump to the front
+                        // of the line to get processed before any other action.
```

Comment:
Do you mean `queue` here?

```suggestion
                        // of the queue to get processed before any other action.
```